yocto-update-strategies.md

```diff
@@ -0,0 +1,651 @@
+  - full build automation in project using Rockchip RK3399
+
+???
+
+Now let's dive into the process of performing updates in Yocto-based projects.
+This stage is crucial for integrating new features, addressing security
+vulnerabilities, and ensuring the system remains up to date. We will discuss
+three key aspects to consider during the update process.
+
+First, the build environment needs to be reviewed and updated. It's important to
+ensure that the build environment is up-to-date and meets the requirements
+specified in the Yocto Project documentation. This ensures compatibility and a
+smooth update process.
+
+Next, we need to update the external layers used in the project. External layers
+provide additional functionality and components to the Yocto-based project. It's
+important to check for updates to these external layers and incorporate the
+latest versions to benefit from bug fixes, security patches, and new features.
+
+Finally, we have the custom layer update. Custom layers are specific to the
+project and may contain project-specific modifications, configurations, or
+recipes. During the update process, we need to review and update these custom
+layers to incorporate necessary changes, improvements, or bug fixes. This
+ensures that the project's customizations remain compatible and functional with
+the updated system.
+
+By addressing these three aspects—build environment, external layers, and custom
+layer update—we can ensure a smooth and successful update process for
+Yocto-based projects. Updating these components in a systematic manner reduces
+the risk of compatibility issues, ensures the project remains up to date, and
+allows for seamless integration of new functionalities and improvements.
+
+---
+# Performing update - build environment
+
+* Verify the compatibility of the build environment with the desired Yocto
+  Project release
+  - see `Build Host Packages` in Mega Manual
+  - use correct release
+* Prepare reproducible build environment
+  - ansible can be used to put together such an environment
+  - other possibility is to prepare containeraized image; e.g.
+    [kas](https://github.com/siemens/kas),
+    [imx-docker](https://github.com/nxp-imx/imx-docker),
+    [yocto-docker](https://github.com/3mdeb/yocto-docker)
+  - easier to distribute such an environment among colleagues
+* Review and update the configuration files (local.conf, bblayers.conf) to
+  reflect any changes or additions required for the update
+  - see release notes, to check if any changes need to be applied
+  - rethink the way build configuration files are stored
+
+???
+
+rethink the way build configuration files are stored - you may want to go back
+to the version before an update
+
+---
+# Performing update - external layers update
+
+.center[<img src="img/project-layers.svg" width="450px">]
+
+* Project consist of external and custom layers
+  - this way we follow the rule **Never modify the POKY layer. Never. Ever.**
```

Comment:
```suggestion
  - this way we follow the rule **Never modify the POKY layer. Never.**
```

Reply:
Removed this and left only an image https://github.com/3mdeb/conferences/pull/7/commits/85813329f1e5bcdea81c19b1f8d9f88cda102040#diff-a1666090e402e433c955cc46797dc399867480b4d815574efd9f5d4181e24ac3R319